File: Этап 1. Структуры/С/Trade.c
#include <stdio.h>
#include <string.h>

// Структура для менеджера
struct Manager {
    int manager_id;
    const char *name;
    const char *city;
    double comm;
};

// Структура для торговли
struct Trade {
    struct Manager managers[100]; // Массив менеджеров
    int manager_count;            // Количество менеджеров
};

// Функция для добавления менеджера в структуру Trade
void add_manager(struct Trade *trade, int manager_id, const char *name, const char *city, double comm) {
    struct Manager man = {manager_id, name, city, comm};
    trade->managers[trade->manager_count++] = man;
}

// Функция для вывода менеджеров из Коврова
void print_kovrov_managers(struct Trade *trade) {
    printf("Ковровские продавцы:\n");
    for (int i = 0; i < trade->manager_count; i++) {
        if (strcmp(trade->managers[i].city, "Ковров") == 0) {
            printf("id=%d, name=%s, city=%s, comm=%.2f\n", 
                   trade->managers[i].manager_id, trade->managers[i].name, trade->managers[i].city, trade->managers[i].comm);
        }
    }
}

int main() {
    struct Trade trade = { .manager_count = 0 };
    
    // Добавляем менеджеров
    add_manager(&trade, 1001, "Иванов", "Ковров", 0.13);
    add_manager(&trade, 1002, "Петров", "Владимир", 0.13);
    add_manager(&trade, 1003, "Симонян", "Москва", 0.10);
    
    // Печатаем менеджеров из Коврова
    print_kovrov_managers(&trade);

    return 0;
}

File: Этап 1. Структуры/trade.py
from client import Client
from manager import Manager
from order import Order

class Trade:
    """

    """
    def __init__(self, trade_id: int):
        """ """
        self.trade_id = trade_id
        self.managers = []
        self.clients = []
        self.orders = []

    def add_manager(self, manager: Manager):
        """ """
        self.managers.append(manager)

    def add_client(self, client: Client):
        """ """
        pass # ToDo

    def add_order(self, order: Order):
        """ """
        pass # ToDo

    def __str__(self):
        # ToDo
        return "?"

    def read_managers(self, managers_fname):
        with open(managers_fname, "r") as managers_f:
            for line in managers_f:
                manager_id, name, city, comm = line.split(";")
                manager_id = int(manager_id)
                comm = float(comm)
                manager = Manager(manager_id, name, city, comm)
                self.managers.append(manager)
    def Kovrov_managers(self):
        """ """
        res = []
        for manager in self.managers:
            if manager.city.upper() == 'КОВРОВ':
                res.append(manager)
        return res

if __name__ == '__main__':
    trade = Trade(1)
    trade.read_managers("manager.txt")
    # ToDo
    # trade.read_clients("client.txt")
    # trade.read_orders("order.txt")
    
    print('Ковровские продавцы')
    print('manager_id name city comm')
    for manager in trade.Kovrov_managers():
        print(manager)
    print()
